server/src/main/java/org/elasticsearch/index/mapper/IgnoreMalformedStoredValues.java

```diff
@@ -155,16 +169,32 @@ private void decodeAndWrite(XContentBuilder b, BytesRef r) throws IOException {
                 case 'i':
                     b.value(new BigInteger(r.bytes, r.offset + 1, r.length - 1));
                     return;
+                case 'j':
+                    decodeAndWriteXContent(b, XContentType.JSON, r);
+                    return;
+                case 's':
+                    decodeAndWriteXContent(b, XContentType.SMILE, r);
+                    return;
                 case 't':
                     if (r.length != 1) {
                         throw new IllegalArgumentException("Can't decode " + r);
                     }
                     b.value(true);
                     return;
+                case 'y':
+                    decodeAndWriteXContent(b, XContentType.YAML, r);
+                    return;
```

Comment:
I decided to patch this "on top" of what we had because we still had to encode the xcontent type. Technically this duplicates the xcontent type - we could store the xcontenttype one time per document, but we are already using a byte here already so it felt ok.

I figured I'd keep the current support for "native" lucene stored fields for int/long/string/whatever. I don't think I'd have built it in the first place if I were building the xcontent copying thing. But because I've already wrote it I didn't think it was worth removing and always using the xcontent copying branch.